pkg/events/derive/hidden_kernel_module.go

```diff
@@ -24,7 +24,7 @@ import (
 )
 
 var (
-	foundHiddenKernModsCache *lru.Cache
+	foundHiddenKernModsCache *lru.Cache[uint64, struct{}]
```

Comment:
Why is this file being changed from v0 to v2 ? Just curious, as we might have many other places using v0 LRU non generics way.

Reply:
Actually, we have "simplelru" instead of pure v0 I believe (symbols_collision, symbols_loaded, host_symbols_loader).

Reply:
I wonder if we should uniformize this. Another thing that occurred to me as well, we have protected generic based maps helper funcions (I believe @AlonZivony created, was it in a non merged PR ?). 

I think we will eventually have to make it uniform:

- mutex protected maps (which is what Alon did IIRC) helper
- same LRU approach in all places (sake of maintainability)
- probably more atomic map like helpers

Reply:
I initially wanted to implement data sources using LRUs as the backend always, but that attempt didn't work out due to generics usage issues.
As part of that commit I checked all places (in tracee open source itself) we use the hashicorp lru and updated it, so I don't think there are any left overs. As I already finished the commit thought I might as well leave it in the PR.

Reply:
> * mutex protected maps (which is what Alon did IIRC) helper

I implemented in my process tree PR (#3054) a RWMap safe to use from multiple go routines.
If you want you can use it (just copy the code to this PR), or use the `sync.map` (which use interfaces instead of generics).
[You can see the code here](https://github.com/aquasecurity/tracee/pull/3054/files#diff-d3fe810a7fe6849ae66f7c3f4e68fc48e877a5cd97a7b8ca726353052d4b70f8)

Reply:
Well a safe RWMap and a LRU cache are not exactly the same thing, I think LRUs are mainly useful when we want to limit the number of items. RWMap has no such syntax. So we likely have a use for both.

Reply:
Either way I will separate this out to a PR for v0.16.0 and @rafaeldtinoco you can commit on top of it.